app/src/main/java/ee/ioc/phon/android/speechutils/AbstractAudioRecorder.java

```diff
@@ -182,7 +257,15 @@ protected void setState(State state) {
      */
     public synchronized byte[] consumeRecording() {
         byte[] bytes = getCurrentRecording(mConsumedLength);
+        // this is to avoid race (set the consumed length to be the recorded length though
+        // the last recording was empty while recorded length moved on - thus we always miss
+        // a part of the recording)
```

Comment:
in reality, given a high enough polling interval, the recording pointer doesn't advance fast enough as the consuming pointer. Thus, there's no new data to process for a given consumeRecording call. To understand it, I'm returning a null buffer (it could be an empty buffer of course, we can discuss this)